test/SlowTests/Tests/Linq/CanQueryAndIncludeRevisions.cs

```diff
@@ -0,0 +1,1982 @@
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var query = await session.Advanced
+                        .AsyncRawQuery<User>("from Users include revisions($p0)")
+                        .AddParameter("p0", "ChangeVector")
+                        .ToListAsync();
+
+                    var revision = await session.Advanced.Revisions.GetAsync<User>(changeVector);
+
+                    Assert.NotNull(revision);
+                    Assert.Equal(1, session.Advanced.NumberOfRequests);
+                }
+            }
+        }
+
+        [Fact]
+        public void Query_RawQueryWithParameters_IncludeRevisions_Array()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var cvList = new List<string>();
+
+                const string id = "users/omer";
+
+                RevisionsHelper.SetupRevisions(Server.ServerStore, store.Database).Wait();
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User {Name = "Omer",},
+                        id);
+
+                    session.SaveChanges();
+                }
+
+                string changeVector;
+                using (var session = store.OpenSession())
+                {
+                    var metadatas = session.Advanced.Revisions.GetMetadataFor(id);
+                    Assert.Equal(1, metadatas.Count);
+
+                    changeVector = metadatas.First().GetString(Constants.Documents.Metadata.ChangeVector);
+
+                    session.Advanced.Patch<User, string>(id, x => x.FirstRevision, changeVector);
+
+                    session.SaveChanges();
+
+                    cvList.Add(changeVector);
+
+                    metadatas = session.Advanced.Revisions.GetMetadataFor(id);
+
+                    changeVector = metadatas[0].GetString(Constants.Documents.Metadata.ChangeVector);
+
+                    cvList.Add(changeVector);
+
+                    session.Advanced.Patch<User, string>(id, x => x.SecondRevision, changeVector);
+
+                    session.SaveChanges();
+
+                    metadatas = session.Advanced.Revisions.GetMetadataFor(id);
+
+                    changeVector = metadatas[0].GetString(Constants.Documents.Metadata.ChangeVector);
+
+                    cvList.Add(changeVector);
+
+                    session.Advanced.Patch<User, string>(id, x => x.ThirdRevision, changeVector);
+
+                    session.SaveChanges();
+
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced
+                        .RawQuery<User>("from Users as u include revisions($p0, $p1, $p2)")
+                        .AddParameter("p0", "u.FirstRevision")
+                        .AddParameter("p1", "u.SecondRevision")
+                        .AddParameter("p2", "u.ThirdRevision")
+                        .ToList();
```

Comment:
This is _super_ strange. You are passing the `u` as the alias here? But that is almost never how this would actually work.